File: R/day_01.R

input_1 <- as.numeric(readLines("day_1/day_1.txt"))

# Part 1 answer
sum(diff(input_1) > 0)

input_roll <- zoo::rollsum(input_1, 3)

# Part 2 answer
sum(diff(input_roll) > 0)

File: python/day_01.py
import pandas as pd

f = open("Coding/AdventOfCode_2021_python/day_1.txt", "r")

input = f.readlines()

f.close()

input = list(map(int, input))

input = pd.Series(input)

# Part 1 answer
print(sum(input.diff(periods = 1) > 0))

input = input.rolling(3).sum()

# Part 2 answer
print(sum(input.diff(periods = 1) > 0))
